routes/account.js

```diff
@@ -611,6 +784,14 @@ module.exports = function (
 
   if (isProduction) {
     delete routes[0].config.validate.payload.preVerified
+
+    for (var i = 0; i < routes.length; ++i) {
+      var route = routes[i];
+      if (route.path === '/account/lock') {
+        routes = routes.splice(i, 1)
```

Comment:
Declarative++.  I wonder whether hapi can cope with holes in its routes array?  Is so then maybe we could do it like:

```
routes = [
  { ...other stuff... },
  isProduction ? undefined : { ...the dev-only route... },
  { ...more stuff... }
```

I also quite like the way that @seanmonstar did it over in the oauth-server, where each route is defined in a separate file and we explicitly stitch together the required routes as a separate step: https://github.com/mozilla/fxa-oauth-server/blob/master/lib/routing.js

Anyway, I don't want to derail this PR, I'd be happy to land with a fixed version of the `splice` technique and file a follow-up to make the whole non-production-code thing safer overall.